### PR TITLE
TypeScript type enhancements and test fixes

### PR DESCRIPTION
**Pull request by mattheyan**
Update `VueModel` to leverage model.js type enhancements.

Also, remove re-exporting of model.js members. They should instead be imported from model.js directly.

### CODE REVIEW

File: package.json
Comment (mattheyan) [inline package.json:1]: _info_: Updated package versions to match what's used in [model.js](https://github.com/cognitoforms/model.js). This addressed some TS and lint issues I encountered.

File: src/index.ts [hunk 1]
```diff
@@ -5,16 +5,6 @@ import "@cognitoforms/model.js/lib/resource-en";
...
-// Export any additional *types* to be referenced externally
-// TODO: Change to export { ... } from "./...";
-export type { Model, ModelOptions, ModelLocalizationOptions } from "@cognitoforms/model.js";
-export type { TypeOptions, TypeExtensionOptions, RuleOrMethodFunctionOrOptions } from "@cognitoforms/model.js";
-export type { Entity } from "@cognitoforms/model.js";
-export type { CultureInfo } from "@cognitoforms/model.js";
-export type { Property, PropertyOptions, PropertyBooleanFunction, PropertyBooleanFunctionAndOptions } from "@cognitoforms/model.js";
-export type { PropertyInjector, PropertySerializationResult } from "@cognitoforms/model.js";
-export { PropertyConverter, IgnoreProperty } from "@cognitoforms/model.js";
-export { isEntityType, isValueType } from "@cognitoforms/model.js";
```
Comment (mattheyan) [inline src/index.ts:17]: _info_: Don't re-export model.js types, functions, etc.

File: src/vue-model.ts [hunk 2]
```diff
@@ -77,8 +77,27 @@ export interface VueModelMixins {
...
+	readonly types: { [T in keyof TTypes]: TypeOfType<TTypes[T]> };
+}
+
+export interface VueModelWithNamespace<TNamespace> extends VueModel {
+	readonly $namespace: ModelNamespace<TNamespace>;
+}
+
+export function createVueModel<TTypes>(options?: ModelTypeOptions<TTypes> & Required<ModelNamespaceOption<TTypes>> & ModelLocalizationOptions, config?: ModelConfiguration): Promise<ModelOfType<TTypes> & ModelWithNamespace<TTypes>>;
+export function createVueModel<TTypes>(options?: ModelTypeOptions<TTypes> & ModelLocalizationOptions, config?: ModelConfiguration): Promise<ModelOfType<TTypes> & ModelNamespace<TTypes>>;
+export function createVueModel<TTypes>(options?: ModelOptions<TTypes>, config?: ModelConfiguration): Promise<ModelOfType<TTypes>> {
```
Comment (mattheyan) [inline src/vue-model.ts:90]: _info_: Add `createVueModel` function to support a typed model

File: tsconfig.json [hunk 3]
```diff
@@ -25,14 +25,15 @@
...
 	],
 	"exclude": [
 		"node_modules",
 		"dist",
 		"build",
 		"out",
 		"lib",
-		"@types"
+		"@types",
+		"./src/**/*.unit.ts"
```
Comment (mattheyan) [inline tsconfig.json:37]: _info_: Don't include unit test files in TS output

**Review by tskimmett — APPROVED** [review]

**Pull request merged**